pkg/storegateway/indexheader/index/postings.go

```diff
@@ -23,17 +24,31 @@ const postingLengthFieldSize = 4
 
 type PostingOffsetTable interface {
 	// PostingsOffset returns the byte range of the postings section for the label with the given name and value.
-	// The Start is inclusive and the End is exclusive.
-	// The end offset might be bigger than the actual posting ending, but not larger than the whole index file.
+	// The Start is inclusive and is the byte offset of the number_of_entries field of a posting list.
+	// The End is exclusive and is typically the byte offset of the CRC32 field.
+	// The End might be bigger than the actual posting ending, but not larger than the whole index file.
 	PostingsOffset(name string, value string) (rng index.Range, found bool, err error)
 
 	// LabelValues returns a list of values for the label named name that match filter and have the prefix provided.
 	LabelValues(name string, prefix string, filter func(string) bool) ([]string, error)
 
+	// LabelValuesOffsets returns all postings lists for the label named name that match filter and have the prefix provided.
+	// The ranges of each posting list are the same as returned by PostingsOffset.
+	LabelValuesOffsets(name, prefix string, filter func(string) bool) ([]PostingListOffset, error)
+
 	// LabelNames returns a sorted list of all label names in this table.
 	LabelNames() ([]string, error)
 }
 
+// PostingListOffset contains the start and end offset of a posting list.
+// The Start is inclusive and is the byte offset of the number_of_entries field of a posting list.
+// The End is exclusive and is typically the byte offset of the CRC32 field.
+// The End might be bigger than the actual posting ending, but not larger than the whole index file.
+type PostingListOffset struct {
+	Value string
```

Comment:
Is this `LabelValue`? I'd consider renaming it, it's not that clear. All this code has a very high cognitive load, and anything we can do to reduce it, is better.